src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -438,12 +438,19 @@ def create_spot_instances(ec2, price, image_id, spec, clusterName,
 
     :rtype: Iterator[list[Instance]]
     """
+    def spotRequestNotFound(e):
+        return e.status == 400 and 'does not exist' in e.body
```

Comment:
I think it would be less brittle to just do `e.error_code == InvalidSpotInstanceRequestID.NotFound`. It would also eliminate the need to check `e.status` since its specific enough.